### ISSUE TITLE
Error Invalid point encoding 0x30 when try to use secp256k1 Key Type

### ISSUE THREAD

**Issue by myrtp** [walt-id/waltid-identity#617]
Hello,

In my Java application, I am using the Walt ID Credentials SDK along with the Issuer and Verifier APIs. I want to use did:key with the secp256k1 KeyType. However, it seems there is a bug with this KeyType because:

1. I can't sign a verifiable presentation:
```
  JsonPrimitive agentVcJwtPrimitive = Json.Default.decodeFromString(JsonPrimitive.Companion.serializer(), vcJwt);
            VerifiablePresentationBuilder verifiablePresentationBuilder = new VerifiablePresentationBuilder();
            verifiablePresentationBuilder.addCredential(agentVcJwtPrimitive);
            verifiablePresentationBuilder.setPresentationId();
            verifiablePresentationBuilder.setDid(subjectDid);
            verifiablePresentationBuilder.setNonce(cryptographyUtils.generateNonce());
            LocalKey localKey = new LocalKey(subjectKeyJwk);
            return verifiablePresentationBuilder.buildAndSign(localKey);
```
I have made the VerifiablePresentationBuilder class that wraps the kotlin  `PresentationBuilder` class.

Also this is the stack trace of the error I get on the walt id library:

```
Java.lang.IllegalArgumentException: Invalid point encoding 0x30
	at org.bouncycastle.math.ec.ECCurve.decodePoint(Unknown Source) ~[bcprov-jdk18on-1.78.jar:na]
	at id.walt.crypto.keys.JvmLocalKeyCreator.ecRawToJwk(LocalKeyGenerator.jvm.kt:50) ~[waltid-crypto-jvm-0.2.0.jar:na]
	at id.walt.crypto.keys.JvmLocalKeyCreator.importRawPublicKey(LocalKeyGenerator.jvm.kt:37) ~[waltid-crypto-jvm-0.2.0.jar:na]
	at id.walt.crypto.keys.LocalKey$Companion.importRawPublicKey(LocalKey.jvm.kt:275) ~[waltid-crypto-jvm-0.2.0.jar:na]
	at id.walt.did.utils.KeyUtils.fromPublicKeyMultiBase-gIAlu-s(KeyUtils.kt:28) ~[waltid-did-jvm-0.2.0.jar:na]
	at id.walt.did.dids.resolver.local.DidKeyResolver.resolveToKey-gIAlu-s(DidKeyResolver.kt:40) ~[waltid-did-jvm-0.2.0.jar:na]
	at id.walt.did.dids.resolver.local.DidKeyResolver.resolve-gIAlu-s(DidKeyResolver.kt:21) ~[waltid-did-jvm-0.2.0.jar:na]
	at id.walt.did.dids.resolver.LocalResolver.resolve-gIAlu-s(LocalResolver.kt:55) ~[waltid-did-jvm-0.2.0.jar:na]
	at id.walt.did.dids.DidService.resolve-gIAlu-s(DidService.kt:138) ~[waltid-did-jvm-0.2.0.jar:na]
	at id.walt.credentials.PresentationBuilder.resolveDidAuthentication(PresentationBuilder.kt:94) ~[waltid-verifiable-credentials-jvm-0.2.0.jar:na]
	at id.walt.credentials.PresentationBuilder.buildAndSign(PresentationBuilder.kt:87) ~[waltid-verifiable-credentials-jvm-0.2.0.jar:na]
```
    

      
2. I can't add credentials to my wallet (running locally).
(screenshot attached)

I have no issues when the KeyType is   `Ed25519`, but I need secp256k1 for compatibility with Ethereum.

I also noticed in your codebase that there is a commented-out test for this, but there is no related issue on GitHub.

```
@JvmStatic
        fun `given raw string, when imported then the import succeeds having the correct key type, key id and hasPrivate values`(): Stream<Arguments> =
            Stream.of(
                ........
                // secp256r1 (throwing Invalid point encoding 0x30)
//                arguments(loadResourceBytes("public-bytes/secp256k1.bin"), KeyType.secp256k1, false),
                // secp256r1 (throwing Invalid point encoding 0x30)
//                arguments(loadResourceBytes("public-bytes/secp256r1.bin"), KeyType.secp256r1, false)
                .........
            )    
  
```

Is there a plan to fix this? Can we help somehow to expedite the process?

Thank you,
Myrto 
![Screenshot_2024-07-12_at_7 57 01_PM](https://github.com/user-attachments/assets/94aa3b80-990d-4108-938c-12e7fff3231a)

**Comment by cpatsonakis** [comment]
Resolved by recently merged PR #630. Please update your dependencies.

Thanks again for your feedback!

**Comment by myrtp** [comment]
The issue was solved by updating the dependencies thank you!
